src/components/ProjectEditView/index.tsx

```diff
@@ -20,9 +28,33 @@ export default function ProjectEditView(): JSX.Element {
     }
   }, [projectId, project, dispatch]);
 
+  const saveProject = useCallback(
+    (_project: UpdateProjectRequest) => {
+      dispatch(requestProjectUpdate({ projectId, project: _project }));
```

Comment:
with the async thunk framework, you can do
```
const { requestId } = dispatch(...);

.. and somewhere above have a useAppSelector/select on the requestId, it should give you a pending | success | error status (you can key off that to show a busy spinner vs error toast message vs goToProject
```

example: [select](https://github.com/terraware/terraware-web/blob/cb7934c20a00c4cfe680255cf43057be09c94d54/src/components/Observations/replacePlot/ReplaceObservationPlotModal.tsx#L44), and handling [here](https://github.com/terraware/terraware-web/blob/cb7934c20a00c4cfe680255cf43057be09c94d54/src/components/Observations/replacePlot/ReplaceObservationPlotModal.tsx#L73) and [busy](https://github.com/terraware/terraware-web/blob/cb7934c20a00c4cfe680255cf43057be09c94d54/src/components/Observations/replacePlot/ReplaceObservationPlotModal.tsx#L137)

Reply:
Alright that's updated. How do I add multiple async thunks to a single slice / reducer? Or am I thinking about it wrong? Eventually I'll need to add the delete request to this reducer, but `extraReducers` doesn't seem to take an array.

https://github.com/terraware/terraware-web/pull/1861/files#diff-a412b6c1013732f27e6157448aadc22fb7716c2f7a944d9dbe7d7734295cb7deR53

Reply:
Or is each request supposed to be a different slice?

Reply:
Here's online [reference](https://redux-toolkit.js.org/api/createAsyncThunk) on createAsyncThunk. There may be a way?
Our current usage has been separate slices by request type.
Current usage for delete is state mapping requestId to status of request. Followed by a dispatch to refresh data.

Reply:
Ended up going with this https://github.com/terraware/terraware-web/blob/main/src/redux/features/projects/projectsSlice.ts#L52-L55

Reply:
Nice 👍